--- evalresp-3.3.0/parse_fctns.c	2009-04-30 22:40:34.000000000 +0200
+++ evalresp-3.3.0_mod/parse_fctns.c	2025-05-20 12:36:45.646723734 +0200
@@ -19,7 +19,6 @@
 #include <stdlib.h>
 
 #include "./evresp.h"
-double atof();
 
 /* parse_pref:  used to parse the prefix from a non-comment RESP file line (returns the blockette
                 number and starting field number for that line as arguments, return values are -1
