--- evalresp-3.3.0/error_fctns.c	2007-02-27 16:00:18
+++ evalresp-3.3.0_mod/error_fctns.c	2024-01-04 14:57:42
@@ -61,7 +61,7 @@
         fprintf(stderr,fmt_str,sval);
         break;
       default:
-        fprintf(stderr,fmt_str);
+        fprintf(stderr, "%s", fmt_str);
         break;
       }
       new_p = strstr(p,fmt_str);
@@ -137,7 +137,7 @@
         fprintf(stderr,fmt_str,sval);
         break;
       default:
-        fprintf(stderr,fmt_str);
+        fprintf(stderr, "%s", fmt_str);
         break;
       }
       new_p = strstr(p,fmt_str);
